src/Microsoft.Azure.SignalR.AspNet/EndpointProvider/ServiceEndpointProvider.cs

```diff
@@ -110,6 +93,24 @@ public string GetServerEndpoint(string hubName)
             return $"{_serverEndpoint}{ServerPath}/?hub={GetPrefixedHubName(_appName, hubName)}";
         }
 
+        public IAccessTokenProvider GetServerAccessTokenProvider(string hubName, string serverId)
+        {
+            if (_accessKey is AadAccessKey aadAccessKey)
```

Comment:
this can be determined in .ctor

Reply:
Why should we do this? The provider has to be generated every time when this method is being called.